Projects/Embark/Sources/Viewables/Actions/SelectAction/EmbarkSelectAction.swift

```diff
@@ -23,68 +72,63 @@ extension EmbarkSelectAction: Viewable {
-                                        .map { link in
-                                            (link ?? defaultLink, result)
-                                        }
-                                }
+                if options.count == 1, let option = options.first {
+                    let button = LoadableButton(
+                        button: Button(
+                            title: option.link.label,
+                            type: .standard(
+                                backgroundColor: .brand(.secondaryButtonBackgroundColor),
+                                textColor: .brand(.secondaryButtonTextColor)
+                            )
+                        )
+                    )
 
-                                return Future((defaultLink, result))
-                            }
-                            .onValue { link, result in
-                                result.keys.enumerated()
-                                    .forEach { offset, key in
-                                        let value = result.values[offset]
-                                        self.state.store.setValue(
-                                            key: key,
-                                            value: value
-                                        )
-                                    }
-
-                                if let passageName = self.state.passageNameSignal.value {
-                                    self.state.store.setValue(
-                                        key: "\(passageName)Result",
-                                        value: result.textValue
-                                    )
-                                }
+                    bag += button.onTapSignal.onValue({ _ in
+                        if option.api != nil {
+                            button.isLoadingSignal.value = true
+                        }
 
+                        handleClick(option: option)?
+                            .onValue({ link in
                                 callback(link)
-                            }
-                    }
-                    if optionsSlice.count < 2, options.count > 1 {
-                        stack.addArrangedSubview(UIView())
+                            })
+                    })
+
+                    bag += view.addArranged(button)
+                } else {
+                    let numberOfStacks =
+                        options.count % 2 == 0
+                        ? options.count / 2 : Int(floor(Double(options.count) / 2) + 1)
+
+                    for iteration in 1...numberOfStacks {
+                        let stack = UIStackView()
+                        stack.spacing = 10
+                        stack.distribution = .fillEqually
+                        view.addArrangedSubview(stack)
+
+                        let optionsSlice = Array(
+                            options[2 * iteration - 2..<min(2 * iteration, options.count)]
```

Comment:
This whole calculation looks complicated here 🧐
Can we simplify this maybe?

Reply:
Ahh, yeah, this was really old code, was able to make this way neater using `chunked` 😄